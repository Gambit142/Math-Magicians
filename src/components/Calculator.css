* {
  box-sizing: border-box;
  margin: 0;
}

body {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;
}

.calculator-div {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  border: 1px solid rgb(116, 116, 116);
}

.display-div {
  padding: 1.2rem;
  align-self: stretch;
  background: #8b8888;
  text-align: end;
  font-size: 1.5rem;
  color: #fff;
}

.button-grid {
  display: -ms-grid;
  display: grid;
  grid-template: repeat(5, 1fr) / repeat(4, 1fr);
}

.button-grid button {
  padding: 2rem;
  font-size: 1rem;
  border: 1px solid #e5c7b2;
}

.zero {
  grid-column: 1 / span 2;
}

.orange-background {
  background-color: #f98538;
}

.navbar {
  overflow: hidden;
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  background-color: #2e1309;
  display: flex;
  justify-content: space-around;
  align-items: center;
}

.navbar h1 {
  font-family: 'Henny Penny', cursive;
  font-size: 2.2rem;
  color: #bfb8b8;
}

.navbar ul {
  display: flex;
  justify-content: space-between;
  list-style: none;
}

.navbar ul li {
  padding: 1.5rem;
}

.navbar ul li a {
  text-decoration: none;
  font-family: 'Quantico', sans-serif;
  font-size: 1.2rem;
  color: #f5f763;
}

.navbar ul li a:active,
.navbar ul li a:focus {
  border: 2px solid #f5f763;
  padding: 0.7rem;
}

.welcome {
  font-family: 'Quantico', sans-serif;
  font-size: 2rem;
  margin: 1rem;
}