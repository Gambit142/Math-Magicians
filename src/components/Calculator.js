import PropTypes from 'prop-types';
import './Calculator.css';

const Calculator = (props) => {
  const { variable } = props;
  const { next, total, operation } = variable;

  const buttonClicked = (e) => {
    const { btn } = props;
    btn(e.target.innerHTML);
  };

  const buttonNames = ['AC', '+/-', '%', '\u00F7', '7', '8', '9', '\u00D7',
    '4', '5', '6', '-', '1', '2', '3', '+', '0', '.', '='];
  const generateClassName = (className) => {
    if (className === '\u00F7' || className === '\u00D7'
    || className === '-' || className === '+' || className === '=') {
      return 'orange-background';
    }
    if (className === '0') {
      return 'zero';
    }
    return '';
  };

  let result = '';
  if (total) {
    result = `${total} ${operation || ''} ${next || ''}`;
  } else if (next) {
    result = next;
  }
  return (
    <div className="calculator-div">
      <div className="display-div">{result || 0}</div>
      <div className="button-grid">
        {buttonNames.map((button) => (<button key={button} className={generateClassName(button)} onClick={buttonClicked} type="button">{button}</button>))}
      </div>
    </div>
  );
};

Calculator.propTypes = {
  btn: PropTypes.func.isRequired,
  variable: PropTypes.shape({
    total: PropTypes.string,
    next: PropTypes.string,
    operation: PropTypes.string,
  }).isRequired,
};

export default Calculator;
