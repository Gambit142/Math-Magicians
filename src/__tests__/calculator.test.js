import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';
import calculate from '../logic/calculate'

describe('Check functionality of Calculator Component', () => {
  it('renders correctly', () => {
    const object = {
      total: '',
      next: '',
      operation: '',
    };

    const updateFunction = (button) => {
      const calculator = calculate(object, button)
    }

    const tree = renderer
      .create(<Calculator variable={object} btn={updateFunction} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Check if Element is in the DOM', () => {
    const object = {
      total: '',
      next: '',
      operation: '',
    };

    const updateFunction = (button) => {
      const calculator = calculate(object, button)
    }

    render(<Calculator variable={object}  btn={updateFunction} />);
    expect(screen.getByText('AC')).toBeInTheDocument();
    expect(screen.getByText('Let\'s Do Some Maths!')).toBeInTheDocument();
    expect(screen.getAllByRole('button')).toHaveLength(19);
  });
});
