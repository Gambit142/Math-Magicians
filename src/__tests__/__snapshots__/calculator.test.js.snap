// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check functionality of Calculator Component renders correctly 1`] = `
<div
  className="calculator-container"
>
  <p>
    Let's Do Some Maths!
  </p>
  <div
    className="calculator-div"
  >
    <div
      className="display-div"
    >
      0
    </div>
    <div
      className="button-grid"
    >
      <button
        className=""
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className=""
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className=""
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="orange-background"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className=""
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className=""
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className=""
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="orange-background"
        onClick={[Function]}
        type="button"
      >
        ร
      </button>
      <button
        className=""
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className=""
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className=""
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="orange-background"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className=""
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className=""
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className=""
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="orange-background"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="zero"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className=""
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="orange-background"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
